model/modules/parsing_ng/README.md

```diff
@@ -37,16 +96,255 @@ capture_or_regex_component = capture_component | regex_component
+extract_part:
+  [condition: regex_component]
+  capture: capture | capture_reference
+extract_parts:
+  extract_part: List<extract_part>
 parsing_component = decomposition | decomposition_cascade |
 ```
 
-# Top-level
+## Explanation
+
+A `decomposition` tries to match an entire string (unless `anchor_beginning` or
+`anchor_end` create exceptions). It can contain either a single `capture` or a
+single `capture_reference`.
+
+Decompositions are embedded in `parsing_definitions` and instruct the browser
+how to process observed data. So for example, if the browser has seen a
+`street-location` field, the following parsing definition can indicate which
+capture to use to process it:
+```yaml
+parsing_definitions:
+  street-location:
+    decomposition:
+      capture_reference: ParseStreetLocation
+```
+
+A `decomposition_cascade` enables us to try one alternative after the next
+until we have found a match. It can even be fitted with a condition to only
+use it in case the condition is fulfilled. Here is a complex case that is used
+for parsing names:
+```yaml
+parsing_definitions:
+  name:
+    decomposition_cascade:
+      alternatives:
+      - decomposition_cascade:
+          condition:
+            regex_reference: kHasCjkNameCharacteristics
+          alternatives:
+          - decomposition: {capture_reference: ParseSeparatedCjkNameExpression}
+          - decomposition: {capture_reference: ParseKoreanTwoCharacterLastNameExpression}
+          - decomposition: {capture_reference: ParseCommonCjkTwoCharacterLastNameExpression}
+          - decomposition: {capture_reference: ParseCjkSingleCharacterLastNameExpression}
+      - decomposition_cascade:
+          condition:
+            regex_reference: kHasHispanicLatinxNameCharacteristics
+          alternatives:
+          - decomposition: {capture_reference: ParseHispanicFullNameExpression}
+      - decomposition_cascade:
+          # No condition, this is the fallback.
+          alternatives:
+          - decomposition: {capture_reference: ParseOnlyLastNameExpression}
+          - decomposition: {capture_reference: ParseLastCommaFirstMiddleNameExpression}
+          - decomposition: {capture_reference: ParseFirstMiddleLastNameExpression}
+```
+If the regex `kHasCjkNameCharacteristics` matches the input, four different ways
+of decompositioning the input are attempted. If neither of those generated a
+match, the next cascade for hispanic/latinx names is attempted.
+
+An alternative strategy is to rely on `extract_parts`. For example a street
+address is inherently complex to parse because apartments, floors, doors, etc.
+can be presented in arbitrary order. `extract_parts` can help looking for
+keywords (e.g. "floor") and adjacent information (e.g. the floor number). Unlike
+for a `decomposition_cascade`, `extract_parts` does not follow the "the first
+match wins" principle but applies all matching attempts in a row.
+
+Here is an example:
```

Comment:
Do I understand correctly that it is not possible to use decompositions and extract_parts within the same rule? If both ways it would be great to reflect the answer in some way. It was not clear to me after reading this.